qrcode_detection_package/main.py

```diff
@@ -1,56 +1,358 @@
-import sys
 import rclpy
+import time
+import subprocess
 from rclpy.node import Node
 import cv2
 from std_msgs.msg import String
 
 from common_package_py.common_node import CommonNode
+from interfaces.msg import QRCodeInfo
+from interfaces.msg import Control
 
 
 class QRCodeScannerNode(CommonNode):
     def __init__(self, id: str):
+        """
+        Constructor of the QRCodeScannerNode class.
+
+        This method initializes the object properties for the QRCodeSearch object.
+        It sets the initial state to "ready" and configures the subscription and publisher objects for control 
+        and QR code information. Additionally, it initializes the QR code detector from OpenCV.
+        The method also includes a configuration variable for selecting the QR code detection method.
+        This method is called automatically when creating a QRCodeSearch object.
+
+        Args: 
+            id: The node_id.
+
+        Returns: 
+            None
+        """
         super().__init__(id)
+        self.set_state("ready")
+        self.control_subscription = self.create_subscription(
+            Control,
+            "control",
```

Comment:
It would be great if you could change the topic names to the brandnew global constants:
https://docs.google.com/document/d/17nq2SS2DX0lrlWp5FjLQez1mpn0i5w_5APbjAiUk3j0/edit#heading=h.a32jkqz88y7w